tools/check_for_upgrade.sh

```diff
@@ -29,31 +29,36 @@ fi
 # Cancel upgrade if git is unavailable on the system
 whence git >/dev/null || return 0
 
-if [ -f ~/.zsh-update ]
```

Comment:
This is perfect. No possibility of race condition here, only one of the processes will get the succesfull return code. :+1: